operator/operator_server.go

```diff
@@ -0,0 +1,118 @@
+package operator
+
+// since operator is a serverless component on its own, its associated server
+// is only used for hosting the /metrics, /livez and /readyz endpoints
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/grafana/grafana-app-sdk/health"
+)
+
+type OperatorServer struct {
+	Checks              []health.Check
+	Mux                 *http.ServeMux
```

Comment:
I think we should avoid exporting the mux if we can here, to avoid handlers from being overwritten.